python/npcomp/torch/opdefs/generate_ods.py

```diff
@@ -0,0 +1,190 @@
+          is_last = index == result_len - 1
+          self.print(f"{value_spec.mlir_ods_predicate}:${value_spec.name}",
+                     end="\n" if is_last else ",\n")
+      self.print(");")
+
+    # Description and extra class declarations.
+    with self.indent():
+      if description:
+        quoted_description = _quote_multiline_docstring(
+            description, indent_level=self.indent_level)
+        self.print(f"let description = {quoted_description};")
+
+    self.print("}\n")
+
+  @contextmanager
+  def indent(self, level=1):
+    self.indent_level += level
+    yield
+    self.indent_level -= level
+    assert self.indent_level >= 0, "Unbalanced indentation"
+
+  def print(self, s, end="\n"):
+    if self.indent_level:
+      self.out.write(_INDENT * self.indent_level)
+    self.out.write(s)
+    self.out.write(end)
+
+
+def _snakecase_to_camelcase(ident: str):
+  return "".join(x.capitalize() or "_" for x in re.split(r"[\._]", ident))
+
+
+def _quote(s: str):
+  s = s.replace(r'"', r'\\"')
+  return f'"{s}"'
+
+
+def _quote_multiline_docstring(s: str, indent_level: int = 0):
+  # Unlikely to contain the delimitter and since just a docstring, be safe.
+  s = s.replace("}]", "")
+  indent = _INDENT * indent_level
+  s = textwrap.indent(s, indent + _INDENT)
+  return "[{\n" + s + indent + "\n" + indent + "}]"
+
+
+def _split_docstring(docstring: str):
+  """Splits the docstring into a summary and description."""
+  lines = docstring.splitlines()
+  # Skip leading blank lines.
+  while lines and not lines[0]:
+    lines = lines[1:]
+  if len(lines) > 2:
+    return lines[0], "\n".join(lines[2:])
+  else:
+    return lines[0]
+
+
+def main(args):
+  r = OpRegistry()
+  # Populate from modules that provide a populate() function.
+  op_modules = [args.op_module]
+  for m_name in op_modules:
+    logging.info(f"Populating from module: {m_name}")
+    m = importlib.import_module(m_name, package=__package__)
+    f = getattr(m, "populate")
+    f(r)
+
+  out = sys.stdout
+
+  # Write file header.
+  module_name = sys.modules["__main__"].__loader__.name
+  out.write("\n".join([
+      "//===-------------------------------------------------------*- tablegen -*-===//",
+      "//",
+      "// This file is licensed under the Apache License v2.0 with LLVM Exceptions.",
+      "// See https://llvm.org/LICENSE.txt for license information.",
+      "// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception",
+      "//",
+      "// Operation summaries and descriptions were systematically derived from public",
+      "// API docstrings and are licensed accordingly:",
+      "//   https://github.com/pytorch/pytorch/blob/master/LICENSE",
+      "//===----------------------------------------------------------------------===//",
+      "// This file is automatically generated.  Please do not edit.",
+      "// Generated via:",
+      f"//   python -m {module_name} {' '.join(sys.argv[1:])}",
```

Comment:
nit: this generates a whitespace. 
<img width="760" alt="Screen Shot 2020-09-15 at 11 40 08 PM" src="https://user-images.githubusercontent.com/74956/93301271-11e75e80-f7ad-11ea-9af6-f66abf100751.png">